core/src/env/traits.rs

```diff
@@ -120,4 +120,6 @@ pub trait Env: EnvTypes {
 
     /// Deposits raw event data through Contracts module.
     fn deposit_raw_event(topics: &[<Self as EnvTypes>::Hash], data: &[u8]);
+
+    fn call(callee : <Self as EnvTypes>::AccountId, gas: u64, value : <Self as EnvTypes>::Balance ,input_data : &[u8])->u32;
```

Comment:
From the documentation of `ext_call`:

```
// Returns 0 on the successful execution and puts the result data returned
// by the callee into the scratch buffer. Otherwise, returns 1 and clears the scratch
// buffer.
```

While the API that returns a `u32` is correct I think even though this is fairly low-level we maybe want a different return type and value.
For example, we might want to return something like an `Option<ScratchBufferAccess>` to indicate that there was or wasn't a result instead of a `u32`.
Through `ScratchBufferAccess` I mean a thin-wrapper type that simply provides access to the returned data if any. This could be a type that simply wraps calls to `ext_scratch_len` and `ext_scratch_copy` and returns `Option<Vec<u8>>`.

As in simply returning `u32` we should avoid copying the result always from the scratch buffer since the claler might not be interested in the actual result and I am not sure if the compiler is smart enough to optimize these calls over FFI away.